app/feature/feature-terminate-insurance/src/main/kotlin/com/hedvig/android/feature/terminateinsurance/step/overview/OverviewViewModel.kt

```diff
@@ -32,7 +33,9 @@ internal class OverviewViewModel(
   val uiState: StateFlow<OverviewUiState> = _uiState.asStateFlow()
 
   fun submitSelectedDate() {
+    _uiState.update { it.copy(isLoading = true) }
     viewModelScope.launch {
+      delay(3000) // Fake delay for better UX
```

Comment:
Let's actually make this a tiny bit nicer next week.
It's good to make it take at least 3 seconds, but not necessarily good to make it take +3 seconds of whatever it may have taken already. Meaning that if the network request actually takes 3 seconds, then we should show this for 3 seconds, not 3 seconds + 3 of the network request = 6 seconds.

We can try doing this by launching the request asap, but only set the response when at least 3 seconds have passed. We can also not show the delay unnecessarily if the request were to fail anyway, and we get the ifLeft response.

Can't quite free-code this in the PR here but we should look into it before we ship this for sure.